src/llama_index_handler.py

```diff
@@ -0,0 +1,279 @@
+                dataset = rg.FeedbackDataset(
+                    fields=[
+                        rg.TextField(name="prompt"),
+                        rg.TextField(name="response"),
+                        rg.TextField(name="time-details", title="Time Details", use_markdown=True),
+                    ],
+                    questions=[
+                        rg.RatingQuestion(
+                            name="response-rating",
+                            title="Rating",
+                            description="How would you rate the quality of the response?",
+                            values=[1, 2, 3, 4, 5],
+                            required=True,
+                        ),
+                        rg.TextQuestion(
+                            name="response-feedback",
+                            title="Feedback",
+                            description="What feedback do you have for the response?",
+                            required=False,
+                        ),
+                    ],
+                        guidelines="You're asked to rate the quality of the response and provide feedback.",
+                        allow_extra_metadata=True,
+                )
+
+                self.dataset = dataset.push_to_argilla(self.dataset_name)
+                warnings.warn(
+                (
+                    f"No dataset with the name {self.dataset_name} was found in workspace "
+                    f"{self.workspace_name}. A new dataset with the name {self.dataset_name} "
+                    "has been created with the question fields `prompt` and `response`"
+                    "and the rating question `response-rating` with values 1-5 and text question"
+                    "named `response-feedback`."
+                ),
+            )
+
+        except Exception as e:
+            raise FileNotFoundError(
+                f"`FeedbackDataset` retrieval and creation both failed with exception `{e}`."
+                f" If the problem persists please report it to {self.ISSUES_URL} "
+                f"as an `integration` issue."
+            ) from e
+        
+        supported_fields = ["prompt", "response", "time-details"]
+        if supported_fields != [field.name for field in self.dataset.fields]:
+            raise ValueError(
+                f"`FeedbackDataset` with name={self.dataset_name} in the workspace="
+                f"{self.workspace_name} had fields that are not supported yet for the"
+                f"`llama-index` integration. Supported fields are: {supported_fields},"
+                f" and the current `FeedbackDataset` fields are {[field.name for field in self.dataset.fields]}."
+            )
+        
+        self.field_names = [field.name for field in self.dataset.fields]
+        self.events_data = {}
+
+    def _log_data(self, event_data: Dict[str, Any]) -> None:
+        """Log data to Argilla."""
+
+        times = f"""
+        Query time: {event_data["query_time"]}
```

Comment:
I think I might be missing tool events in this overview. 

![image](https://github.com/argilla-io/argilla-llama-index/assets/25269220/ccda01f9-be54-4aba-8a1b-f1b0ec7e62c9)